Information Architecture v3

mDuo13 opened the pull request:
This continues the work from the IA.v2 branch (PR #1592) but addresses issues that came up in reconciling changes with the master branch

oeggert committed:
Update look up escrows to remove redundant info about lookups via sender/destination. Modify cancel expired escrow for brevity.

oeggert committed:
Add draft of updated cancel-escrow.js.

oeggert committed:
Update intro to escrows.

DennisDawson committed:
Add Escrow Tutorial

DennisDawson committed:
Minor corrections

DennisDawson committed:
Fix headings, add HTML

oeggert committed:
Concept info from send-a-time-held-escrow added to escrow.md

DennisDawson committed:
Moving use cases

oeggert committed:
Clean up checks conceptual info.

oeggert committed:
Remove link to checks amendement.

oeggert committed:
Add note to account_objects.md about commandline interface type field.

oeggert committed:
Merge expiration case with lifecycle section.

DennisDawson committed:
Fleshing out Use Cases

Note, the dactyl-config.yml file has not been fully updated.

DennisDawson committed:
Interoperability Use Cases

DennisDawson committed:
Add graphics to intro

oeggert committed:
Move escrow use cases to dedicated page.

oeggert committed:
Update use case page intros and corresponding concept info.

oeggert committed:
Clarify meaning of direct XRP payments.

DennisDawson committed:
Intro link updates

DennisDawson committed:
Add NFT mkt use case

oeggert committed:
Payment use cases

mDuo13 committed:
p2p payments: edits to Wallets

oeggert committed:
Clean up payments use cases

maria-robobug committed:
Move rippled and clio tutorials into infrastructure

DennisDawson committed:
Refine history description

mDuo13 committed:
Touch up stablecoin issuer use case (#1856)

* Consolidate stablecoin use case

* Stablecoin issuer: cleanup progress through sending

* Stablecoin issuer: reorg second half

(Note: the dactyl-config.yml is not fully reconciled yet)

mDuo13 committed:
Cancel escrow: fix notes

mDuo13 committed:
Update escrow docs

This commit re-creates f205a92db217b2f870634c1178103f6705fffa48 with
some adjustments:

- Omit the accidentally-created dir full of junk
- Fix some typos and one mistake in the Escrow limitations section
- Add a table to the EscrowCreate ref to clarify valid combos of fields.

mDuo13 committed:
IA: Move "Consensus Network" files

This re-creates some work from the original commit 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

tequdev committed:
fix Absolute path link to internal content

mDuo13 committed:
Rewrite escrows article (re-created)

This commit re-creates relevant work from the following commits:

9a4a588f2bf4490cbe8d1929f1521043a9a7016d Update escrow.md context info
e1b017dc83d32c4bb3961508cb8d9ba07e543aab Remove references to using escrow for interledger payments.

mDuo13 committed:
IA: Move "XRPL servers" files

This re-creates some work from original commit 7611979abfc72bbdff8d3d4e27fde1477df4cb1d

mDuo13 committed:
IA: move "production readiness" files.

Re-creates work from the following commit:

692438693ac6e3141bccdcd7472e32a85786e975  Move tutorials to concepts

mDuo13 committed:
New intro articles

Original commit: 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

mDuo13 committed:
IA: Reorg account concepts

Re-creates some work from original commit 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

mDuo13 committed:
IA: reorg transaction concepts

Original commits:
9d4eff994012e1e103821604962d0a566e6a8f52  WIP - reorg accounts
7611979abfc72bbdff8d3d4e27fde1477df4cb1d  WIP dir. reorg

mDuo13 committed:
IA: reorg consensus concepts

Original commit: 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

mDuo13 committed:
IA: Reorg ledger docs

Original commit: 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

- Rephrased some details of the section

mDuo13 committed:
IA: rename issuing/operational addresses page

Original commit: 56fffe0b9fae6c5c78c09f6b60a0040655bcf61a

mDuo13 committed:
Remove redundant checks use case section

Original commit: 3c29e9c05e2b81b2aba2c376711d5a9d6771f6f5

mDuo13 committed:
IA: move Dex under tokens

Original commit: d08b3ba7d78e0d324a6db099ebb10637ccaceb5a

mDuo13 committed:
Remove some unnecessary links in transactions section

Original commit: e6fcf4a4dcb43b73395f439da312cf87ecd864fc

mDuo13 committed:
Link cleanup in Tokens section

Original commit: 9588dd5e706431f970e178f19a44ed931f7e5de3

mDuo13 committed:
Touch up 'Configure Peering' section

Original commit: fc8f0990b8bb7863dfc4928d2450fd8b116d2fa9

mDuo13 committed:
Clean up links in accounts section

Original commit: 3da5fde7a8251756d4137a5d1b6a3e7746c5f55b

mDuo13 committed:
IA: use case cleanup

mDuo13 committed:
IA: reconcile servers, ledgers sections

mDuo13 committed:
IA: reconcile payment types, tx, tokens

mDuo13 committed:
IA: reconcile accounts section

mDuo13 committed:
IA: reconcile infra

mDuo13 committed:
IA: Fix most broken links

mDuo13 committed:
Full Docs Index: omit from sidebar

mDuo13 committed:
IA: fix up most broken links

mDuo13 committed:
Quick updates to Software Ecosystem

mDuo13 committed:
Remove some absolute links to internal resources

mDuo13 committed:
Fix remaining broken links in JA target

mDuo13 committed:
Contributing: tweak formatting

mDuo13 committed:
Tutorials: fix some minor issues

mDuo13 commented:
Rebased and force-pushed to fix the merge conflict. Hopefully this will be the last.

oeggert committed:
remove interop use cases

oeggert committed:
remove intro image and personal references to dennis

oeggert committed:
alphabetize-transaction-nav

mDuo13 committed:
Add tooling to check for file/nav consistency

- From the repo top, run tool/check_file_consistency.py to look for
  Markdown files that exist in the "content/" directory but aren't used
  in the documentation.
- New "enforce_filenames" filter prints a warning to console when
  building, if a file's path and filename don't match expectations
  based on its place in the nav and top heading.

oeggert committed:
Merge pull request #1943 from XRPLF/remove-interop-uc

remove interop use cases from IA.v3

oeggert committed:
Merge pull request #1944 from XRPLF/xrp-intro-fix

remove intro image and personal references to dennis on IA.v3

oeggert committed:
Merge pull request #1945 from XRPLF/fix-transaction-nav

alphabetize-transaction-nav in IA.v3

mDuo13 committed:
Remove unused files

mDuo13 committed:
IA: move ledgers, consensus protocol files around

mDuo13 committed:
IA: update nav for new page hierarchy

mDuo13 committed:
Ledger structure: rewrite for accuracy and clarity

DennisDawson committed:
Add QS escrow tutorials

mDuo13 committed:
Update links to removed 'tree format' header

oeggert committed:
reordering of topics under new networks and servers top-nav

DennisDawson committed:
Move "Naming" to "What is XRP?"

DennisDawson committed:
Delete xrp.md

DennisDawson committed:
Update dactyl-config.yml

Remove xrp.md from the TOC.

DennisDawson committed:
Update list-xrp-as-an-exchange.md

Update link to what-is-xrp

DennisDawson committed:
Update list-xrp-as-an-exchange.ja.md

Change link to what-is-xrp

DennisDawson committed:
Update currency-formats.md

Change link to what-is-xrp

DennisDawson committed:
Update currency-formats.ja.md

Change link to what-is-xrp

DennisDawson committed:
Update cancel-an-expired-escrow.md

Change link to what-is-xrp

DennisDawson committed:
Update paymentchannelfund.md

Change link to what-is-xml

DennisDawson committed:
Update look-up-escrows.md

Change link to what-is-xrp

DennisDawson committed:
Update tokens.md

change link to what-is-xrp

DennisDawson committed:
Update use-payment-channels.md

DennisDawson committed:
Update send-a-time-held-escrow.md

Update link to what-is-xml

oeggert committed:
fix broken links

DennisDawson committed:
Update parallel-networks.md

Change link to what-is-xml

DennisDawson committed:
Update parallel-networks.ja.md

DennisDawson committed:
Update invariant-checking.md

Remove link to xrp.html

DennisDawson committed:
Update invariant-checking.ja.md

Remove link to xrp.html

DennisDawson committed:
Update transaction-cost.md

Change link to what-is-xrp

DennisDawson committed:
Update transaction-cost.ja.md

Change link to what-is-xrp

DennisDawson committed:
Update send-a-conditionally-held-escrow.md

Change link to what-is-xrp

DennisDawson committed:
Update stablecoin-issuer.md

Change link to what-is-xrp

DennisDawson committed:
Update tokens.ja.md

Change link to what-is-xml

DennisDawson committed:
Update autobridging.ja.md

Change link to what-is-xrp

DennisDawson committed:
Update currency-formats.md

update text

oeggert committed:
reorganize infrastructure nav section

DennisDawson committed:
Update currency-formats.md

Try removing link altogether.

DennisDawson committed:
Update currency-formats.ja.md

Remove link to what-is-xrp.html

oeggert committed:
move commandline usage topic to infrastructure

oeggert committed:
initial intro rewrite

oeggert committed:
minor update to language

oeggert committed:
Merge pull request #1957 from XRPLF/rewrite-xrpl-intro

initial intro rewrite

mDuo13 committed:
Merge pull request #1950 from XRPLF/rename-servers-nav

Reorder topics under new "Networks and Servers" and "Infrastructure" top-nav.

DennisDawson committed:
Merge pull request #1954 from XRPLF/Consolidate_XRP_pages

Move "Naming" to "What is XRP?"

mDuo13 committed:
IA.v3: rm Production Readiness

DennisDawson committed:
Merge branch 'IA.v3' of github.com:XRPLF/xrpl-dev-portal into IA.v3
Fast forward

mDuo13 committed:
Update xrp link in snippet

mDuo13 committed:
Add redirect for old xrp.html URL

mDuo13 committed:
Small edits to 'What is XRP?' article

mDuo13 committed:
Merge pull request #1951 from XRPLF/ledger_reorg_only

IA.v3: Ledger reorg only

mDuo13 committed:
Merge pull request #1959 from XRPLF/rm_production_readiness

IA.v3: Remove now-unused Production Readiness landing

DennisDawson committed:
Merge branch 'IA.v3' of github.com:XRPLF/xrpl-dev-portal into IA.v3
I don't want to commit this.

DennisDawson committed:
Merge branch 'IA.v3' of github.com:XRPLF/xrpl-dev-portal into RemoveXrpFromTOC
Fixing a conflict.

DennisDawson committed:
Merge branch 'IA.v3' into RemoveXrpFromTOC
Fixing a conflict.

DennisDawson committed:
Add missing imgs

DennisDawson committed:
Merge pull request #1956 from XRPLF/RemoveXrpFromTOC

Remove old XRP page in favor of new "What is XRP?" concept

mDuo13 committed:
XRP - copy edit per @DennisDawson

oeggert committed:
restructure tutorials nav and pages

oeggert committed:
fix broken links

oeggert committed:
more broken link fixes

oeggert committed:
Merge pull request #1986 from XRPLF/update-tutorials-nav

restructure tutorials nav and pages

mDuo13 committed:
Merge pull request #1960 from XRPLF/what_is_xrp_copyedits

Minor copy-edits to "What is XRP?" article

mDuo13 committed:
Algo trading: 1st draft

mDuo13 committed:
Algo trading: notes on taxes

mDuo13 committed:
Merge pull request #1946 from mDuo13/rm_unused_files

Remove unused files

mDuo13 committed:
Algo trading: edits per review

mDuo13 committed:
algo trading: fix broken link

mDuo13 committed:
Merge pull request #1990 from XRPLF/dex_trade_uc

IAv3: Algorithmic trading use case

mDuo13 committed:
Merge branch 'master' into IA.v3

mDuo13 committed:
Ledger Structure: Update diagrams

mDuo13 committed:
Re-gen CSS for ledger structure changes

mDuo13 committed:
Ledger structure: edits per review

mDuo13 committed:
IA.v3: fix broken NFT links introduced by rebase

mDuo13 committed:
Merge pull request #1947 from XRPLF/fixup_ledger_structure

IAv3: Update Ledger Structure page

mDuo13 committed:
Desktop Wallet (py): update little stuff

mDuo13 committed:
Update some capacity/storage details

oeggert committed:
contribute doc nav update

oeggert committed:
fix image link in create diagram page

mDuo13 committed:
IAv3: Fix 'Ledgers' blurb

mDuo13 committed:
Update full history requirements with details from community members

oeggert committed:
add reviewer suggestions

oeggert committed:
Merge pull request #2013 from XRPLF/contribute-docs

contribute doc nav update

mDuo13 committed:
Merge pull request #2015 from mDuo13/fix_ledger_blurb

IAv3: Fix incomplete 'Ledgers' blurb

mDuo13 committed:
Merge branch 'master' into IA.v3

mDuo13 committed:
Edits per @trippled review

mDuo13 committed:
Apply suggestions from peer review

Co-authored-by: oeggert <117319296+oeggert@users.noreply.github.com>

mDuo13 committed:
FH: reword file size limit note per review

DennisDawson committed:
Update software ecosystem

mDuo13 committed:
Merge pull request #2012 from XRPLF/iav3_edits

[IA.v3] Update miscellaneous out-of-date stuff

DennisDawson committed:
updates per review

DennisDawson committed:
Minor tweaks to graphics

DennisDawson committed:
fixTypos

DennisDawson committed:
Update content/concepts/introduction/software-ecosystem.md

Co-authored-by: Amarantha Kulkarni <amarantha-k@users.noreply.github.com>

DennisDawson committed:
Update content/concepts/introduction/software-ecosystem.md

Co-authored-by: Amarantha Kulkarni <amarantha-k@users.noreply.github.com>

DennisDawson committed:
Merge pull request #2019 from XRPLF/fixTypos

Fix typos

mDuo13 committed:
Merge branch 'master' into IA.v3

tequdev commented:
The following file isn't used anywhere in `dactyl-config.yml`
`content/concepts/introduction/xrp.md`

---
this will resolve
- https://github.com/XRPLF/xrpl-dev-portal/pull/2039

Commit:
[JA] update AccountDelete cost

DennisDawson committed:
Merge pull request #2017 from XRPLF/softwareES

Update software ecosystem

mDuo13 committed:
Merge pull request #2030 from develoQ/IA.v3-JA-txn-cost

[JA] Update txn cost

oeggert committed:
custom transactors doc

oeggert committed:
add doc to dactyl config

Commit:
[JA] fix NonFungibleTokensV1_1 amendment status

Commit:
[JA] update NFTokenOffer page

amarantha-k committed:
Update to reflect current UNL publishers

mDuo13 committed:
Remove old, unused XRP article (#2039)

Silkjaer commented on content/concepts/introduction/what-is-xrp.md:
```diff
@@ -0,0 +1,65 @@
+
+![Man with one XRP](img/introduction13-x-prefix.png)
+
+At the time of its creation, there were 100 billion XRP. XRP are held in escrow and sold at intervals that help to maintain the stability of the currency and ensure that the general supply grows predictably.
```
This should be elaborated. XRP is held in escrow, but released to whom and is the sales only to help maintain the stability of the currency?

Silkjaer commented on content/concepts/introduction/what-is-xrp.md:
```diff
@@ -0,0 +1,65 @@
+
+![One Hundred Billion with a "B"](img/introduction14-hundred-billion.png)
+
+### Naming
```
Should add a paragraph about trademarks.

Silkjaer reviewed (changes requested):
Thanks for this great update, it's overall a fantastic rearrangement and additions to the already good documentation. I added a few notes, and have shared preview links with a few others to get some external feedback that i'll be adding in a few days (if any).

oeggert committed:
add reviewer suggestions

mDuo13 committed:
Merge pull request #2040 from develoQ/ja-fix-nftokenoffer

[JA] Update NFTokenOffer page

oeggert committed:
Merge pull request #2034 from XRPLF/custom-transactors

Custom Transactors doc to IA.v3

tequdev commented on content/contributing/documentation-languages.md:
```diff
@@ -0,0 +1,74 @@
+
+## Where to Start
+
+If you want to translate the XRP Ledger Dev Portal into your native language of choice, start with the [XRP Ledger Overview file](https://github.com/XRPLF/xrpl-dev-portal/blob/master/content/concepts/introduction/xrp-ledger-overview.md), which describes the core properties and functions of the XRP Ledger.
```
When this PR is merged into the master branch, `content/concepts/introduction/xrp-ledger-overview.md` will not exist in the branch.

mDuo13 committed:
File consistency checker: correctly handle filenames starting in _

mDuo13 committed:
Remove unused old 'get started' and associated code

mDuo13 committed:
Create Resources section & reorg some files

- Rename some files/folders based on their place in the nav
- Move a bunch of non-documentation stuff, and docs on contributing code
  and/or docs to the new "Resources" section.
- Known issue: nav spills into a second row on page widths between
  993px-1110px. To be fixed in a later CSS update, maybe along with
  making the Resources dropdown multi-column.

mDuo13 committed:
Fix #2078 code tab bug

CSS not built yet, to reduce merge conflicts. Won't have any effect
until that happens.

tequdev committed:
fix Transaction JSON

tequdev committed:
[JA] translate contributing contents

tequdev committed:
fix contributing-to-documentation parent

tequdev committed:
fix contribute-code blurb

mDuo13 committed:
Top nav: add cols for Resources, fix broken links

mDuo13 committed:
CSS: fix top nav overflows

mDuo13 committed:
Fix broken link from redirect not in JA target

mDuo13 committed:
Top nav: add Infra to article types

mDuo13 committed:
Merge pull request #2079 from XRPLF/IAv3_top_nav

[IAv3] Top nav and other fixes

mDuo13 committed:
Merge pull request #2080 from develoQ/fix-tx-req-json

fix Transaction JSON

mDuo13 committed:
Update contrib info & rename intro file

mDuo13 committed:
Merge branch 'ja-IA.v3-contributing' into IA_v3_tl_cleanup

mDuo13 committed:
[ja] Update link to suggested first page to translate

mDuo13 committed:
[ja] fix contribute docs organization

maria-robobug committed:
Run private network with docker tutorial (#2065)

* [NO-ISSUE] Run private network with docker tutorial

Adds a tutorial page in the Infrastructure section on how to run a private XRPL network with Docker.

Please let me know if you think this is a useful page to include for developers, whether the steps are clear or not, and if you have suggestions on what can be added to it.

* Add minor link fixes and Japanese target

* Apply suggestions from code review

Co-authored-by: Amarantha Kulkarni <amarantha-k@users.noreply.github.com>

* Add link to ripple-docker-testnet setup scripts in See Also section

* Update repo URL

---------

Co-authored-by: Amarantha Kulkarni <amarantha-k@users.noreply.github.com>

DennisDawson committed:
add intro gfx (#2036)

* add intro gfx

* Move graphic up

* Update some graphics with their revised versions

* Add updated version of the custodial vs non-custodial graphic

---------

Co-authored-by: Amarantha Kulkarni <amarantha-k@users.noreply.github.com>
Co-authored-by: Amarantha Kulkarni <akulkarni@ripple.com>

amarantha-k committed:
Resolved merge conflict

mDuo13 committed:
[ja] update contributing

Co-authored-by: tequ <git@tequ.dev>

mDuo13 committed:
Merge pull request #2077 from mDuo13/IA_consistency_tooling

Add tooling to check for file/nav consistency

mDuo13 committed:
Merge pull request #2091 from XRPLF/IA_v3_tl_cleanup

Clean up contributing info

amarantha-k committed:
Incorporate feedback on "What is XRP" page. (#2099)

* Add trademark info for XRP

* Revert section to previous state

mDuo13 commented:
@develoQ I believe the changes you requested have been addressed by #2091.
@Silkjaer I believe the changes you requested have been addressed by #2099.

mDuo13 committed:
Merge branch 'master' into IA.v3

tequdev reviewed (changes requested):
<img width="729" alt="スクリーンショット 2023-09-01 13 28 30" src="https://github.com/XRPLF/xrpl-dev-portal/assets/69445828/8e0d3121-840d-49ec-aa1c-312fed0a81bc">

Error logs are being output at build time that were not there before.

 I will now check other Japanese-related sections, including #2091.

Silkjaer reviewed (approved):
LGTM content-wise and ready to merge if the issues @develoQ mentioned are fixed.

tequdev reviewed (approved):
There were no serious problems with the Japanese pages.

I have made some fixes to improve side-nav,top-nav for japanese.

If this PR will be merged into the main branch soon, I'll create a PR for main branch after the merge.
If it will not be merged for a while, I will create a PR targeting IA.v3 branch.

amarantha-k commented:
Yes, we plan to merge soon. 
+1 to creating a PR for the main branch after the merge. 
Thanks for your contribution!

amarantha-k committed:
Fix broken link (#2101)
